tests/integration.rs

```diff
@@ -4271,12 +4271,18 @@ mod parallel {
     }
 
     #[test]
-    #[cfg(target_arch = "x86_64")]
     fn test_cpu_hotplug() {
-        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        #[cfg(target_arch = "aarch64")]
+        let focal_image = FOCAL_IMAGE_UPDATE_KERNEL_NAME.to_string();
+        #[cfg(target_arch = "x86_64")]
+        let focal_image = FOCAL_IMAGE_NAME.to_string();
+        let focal = UbuntuDiskConfig::new(focal_image);
         let guest = Guest::new(Box::new(focal));
         let api_socket = temp_api_path(&guest.tmp_dir);
 
+        #[cfg(target_arch = "aarch64")]
+        let kernel_path = edk2_path();
```

Comment:
Out of curiosity, why aren't you using the direct kernel boot method here? As long as you know the guest kernel you built contains the expected patches, we shouldn't need to make a difference between x86 and aarch64.
That would also simplify things with the image as you wouldn't need a special disk image for this test.

Reply:
Hello, VCPU hotplug needs ACPI support which depends on firmware boot on aarch64, thus we need boot from edk2 here. And I change the disk image by referring to  "test_memory_hotplug".